src/main/java/com/team4/testingsystem/repositories/TestsRepository.java

```diff
@@ -129,4 +129,9 @@
             + "and t.isAvailable = true ")
     boolean hasStartedTests(Long userId);
 
+    @Transactional
+    @Modifying
+    @Query(value = "update Test t set t.listeningAttempts = ?1 where t.id = ?2 ")
+    void updateListeningAttempts(Integer attempts, Long testId);
```

Comment:
potentially, 2 threads can decrease the attempt in parallel.
Let's also check the previous value was not changed, and retry, if the update failed